src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newModifiedArr = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(currentState, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete currentState[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      currentState = {};
+    }
+
+    newModifiedArr.push({ ...currentState });
```

Comment:
[DONT REPEAT YOURSELF] - If you perform the same action in all `switch` cases - do it just once afterwards. To avoid repetition, move the `newModifiedArr.push({ ...currentState });` outside of the `switch` block, after all cases have been processed.